e2e/framework/framework.go

```diff
@@ -50,6 +116,10 @@ type Environment struct {
 
 // New creates a Framework
 func New() *Framework {
+	flag.Parse()
```

Comment:
Our flag parsing is pretty complex (and there are TODOs to add more validation), it might be time to create a flag parser (which we can test), call parse on that and provide a resulting config object into `New`. Just a thought for later